### PR TITLE
Core, Spark: Avoid manifest copies when importing data to V2 tables

### PR DESCRIPTION
**Pull request by aokolnychyi**
This PR extends the idea from #8928 to `FastAppend` and `MergeAppend`, which are used in data imports and migration.

### CODE REVIEW

File: api/src/main/java/org/apache/iceberg/AppendFiles.java [hunk 1]
```diff
@@ -42,16 +42,17 @@ public interface AppendFiles extends SnapshotUpdate<AppendFiles> {
    * <p>The manifest must contain only appended files. All files in the manifest will be appended to
    * the table in the snapshot created by this update.
    *
-   * <p>By default, the manifest will be rewritten to assign all entries this update's snapshot ID.
-   * In that case, it is always the responsibility of the caller to manage the lifecycle of the
-   * original manifest.
+   * <p>The manifest will be used directly if snapshot ID inheritance is enabled (all tables with
```
Comment (aokolnychyi) [inline api/src/main/java/org/apache/iceberg/AppendFiles.java:45]: Simply clarifies the notion of snapshot ID inheritance and that it is always on in V2 tables.

File: api/src/main/java/org/apache/iceberg/AppendFiles.java [hunk 2]
```diff
@@ -42,16 +42,17 @@ public interface AppendFiles extends SnapshotUpdate<AppendFiles> {
...
+   * Otherwise, the manifest will be rewritten to assign all entries this update's snapshot ID.
    *
-   * <p>If manifest entries are allowed to inherit the snapshot ID assigned on commit, the manifest
-   * should never be deleted manually if the commit succeeds as it will become part of the table
-   * metadata and will be cleaned up on expiry. If the manifest gets merged with others while
-   * preparing a new snapshot, it will be deleted automatically if this operation is successful. If
-   * the commit fails, the manifest will never be deleted and it is up to the caller whether to
-   * delete or reuse it.
+   * <p>If the manifest is rewritten, it is always the responsibility of the caller to manage the
+   * lifecycle of the original manifest. If manifest entries are allowed to inherit the snapshot ID
```
Comment (szehon-ho) [inline api/src/main/java/org/apache/iceberg/AppendFiles.java:50]: What do you think to simplify 'if manifest entries are allowed to inherit the snapshot ID...' to 'if the manifest is used directly'..?

**Review by szehon-ho — APPROVED** [review]
looks good to me, minor comment

File: api/src/main/java/org/apache/iceberg/AppendFiles.java (hunk 2 above)
Comment (aokolnychyi) [reply api/src/main/java/org/apache/iceberg/AppendFiles.java]: Yep, I liked it. Fixed.

**Comment by aokolnychyi** [comment]
Thanks, @szehon-ho!

**Pull request merged**